packages/runtime/runtime-utils/src/dataStoreRegistry.ts

```diff
@@ -0,0 +1,62 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import {
+    FluidDataStoreRegistryEntry,
+    IFluidDataStoreRegistry,
+    FluidDataStoreRegistry,
+    FluidDataStoreRegistryEntries,
+    IProvideFluidDataStoreRegistry,
+} from "@fluidframework/runtime-definitions";
+
+export function createDataStoreRegistry(arg: FluidDataStoreRegistry): IProvideFluidDataStoreRegistry {
+    // Ensure that FluidDataStoreRegistry type did not change from original
+    // We do casting below and need to ensure that only these two types can come in
+    const entries: FluidDataStoreRegistryEntries | IProvideFluidDataStoreRegistry = arg;
+
+    const registry = entries as IProvideFluidDataStoreRegistry;
+    if (registry.IFluidDataStoreRegistry !== undefined) {
+        return registry;
+    }
+    return new DataStoreRegistry(entries as FluidDataStoreRegistryEntries);
+}
+
+class DataStoreRegistry implements IFluidDataStoreRegistry {
+    private readonly map: Map<string, FluidDataStoreRegistryEntry | Promise<FluidDataStoreRegistryEntry>> = new Map();
+
+    public get IFluidDataStoreRegistry() { return this; }
+
+    constructor(namedEntries: FluidDataStoreRegistryEntries) {
+        for (const entry of namedEntries) {
+            if (Array.isArray(entry)) {
+                this.map.set(entry[0], entry[1]);
+            } else {
+                this.map.set(entry.IFluidDataStoreFactory.type, entry);
+            }
+        }
+    }
+
+    public async get(name: string): Promise<FluidDataStoreRegistryEntry | undefined> {
```

Comment:
Playing with code today and realized that we actually need this to be get(string[]) not get(string) because of subregistries. Components in subregistries are created from an array of strings and this is what componentContext.packagePath etc. returns so that's what all internal registry logic will use (including ours)

Reply:
Also slightly related - because implementing one may(?) help with the other - the full list of loaded registry entries including loaded subregistry entries needs to be iterable so that we can actually use the registry to pull manifest entries for all components registered subregistry or not.

@anthony-murphy is probably more familiar with where this subregistry logic is located today so it can be combined with this implementation.

Reply:
First, this is comment / suggestion is unrelated to this PR. If we want to have that discussion, we should open an issue for that.
But on substance, what you are asking for is an utility layer on top of registry, that can take an array and do multi-level drill in.

Reply:
Let me open an issue for this then :) I believe that this utility layer already exists and just needs to be made available but we can discuss on the issue

Reply:
https://github.com/microsoft/FluidFramework/issues/4472